pkg/taxonomy/module_schema.json

```diff
@@ -0,0 +1,72 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "https://github.com/IBM/the-mesh-for-data/pkg/taxonomy/module_schema.json",
+    "title": "Module Taxonomy",
+    "description": "Portion of the taxonomy contributed to by the mesh for data modules.",
+    "definitions": {
+        "credential_management": {
+            "type": "string",
+            "description": "The method used by the M4D manager to handle data and user credentials.",
+            "enum": ["secret-provider", "automatic"]
+        },
+        "module_type": {
+            "type": "string",
+            "description": "Type of modules supported by the M4D manager.",
+            "enum": ["copy", "read", "write"]
+        },
+        "module_dependency": {
+            "type": "object",
+            "description": "Process on which a module is dependent.",
+            "properties": {
+                "dependency_type": {
+                    "type": "string",
+                    "description": "The type of processes on which a module can have dependencies.",
+                    "enum": ["module", "connector", "feature"]
+                },
+                "name": {
+                    "type": "string"
+                }
+            }
+        },
+        "protocol": {
+            "type": "string",
+            "description": "Protocol via which data is accessed.",
+            "enum": ["s3", "kafka", "jdbc-db2", "m4d-arrow-flight"]
```

Comment:
I don't think protocol should be an enum.
The list of supported protocols by M4D could be build at run-time by collecting the list of supported protocol from the installed modules, and that list can generate checkers that M4D would use to validate M4DApplication input.

Reply:
See my comment below about owning systems